hazelcast/src/main/java/com/hazelcast/config/XmlConfigBuilder.java

```diff
@@ -649,7 +649,26 @@ private void handleScheduledExecutor(Node node) throws Exception {
 
     private void handleCardinalityEstimator(Node node) throws Exception {
         CardinalityEstimatorConfig cardinalityEstimatorConfig = new CardinalityEstimatorConfig();
-        handleViaReflection(node, config, cardinalityEstimatorConfig);
+        cardinalityEstimatorConfig.setName(getTextContent(node.getAttributes().getNamedItem("name")));
+
+        for (Node child : childElements(node)) {
+            String nodeName = cleanNodeName(child);
+            if ("merge-policy".equals(nodeName)) {
+                MergePolicyConfig mergePolicyConfig = new MergePolicyConfig();
```

Comment:
You can use `com.hazelcast.config.XmlConfigBuilder#createMergePolicyConfig` as `handleAtomicLong` and `handleAtomicReference` do